ui/apps/platform/src/Containers/Collections/RuleSelector/RuleSelector.cy.jsx

```diff
@@ -0,0 +1,253 @@
+import React, { useEffect, useState } from 'react';
+import isEqual from 'lodash/isEqual';
+
+import RuleSelector from './RuleSelector';
+
+// Component wrapper to allow a higher level component to feed updated state back to the RuleSelector.
+function DeploymentRuleSelector({ defaultSelector, onChange }) {
+    const [resourceSelector, setResourceSelector] = useState(defaultSelector);
+
+    return (
+        <RuleSelector
+            entityType="Deployment"
+            scopedResourceSelector={resourceSelector}
+            handleChange={(_, newSelector) => {
+                setResourceSelector(newSelector);
+                onChange(newSelector);
+            }}
+            validationErrors={undefined}
+        />
+    );
+}
+
+function setup(defaultSelector, onChange) {
+    cy.intercept('GET', '/v1/collections/*/autocomplete', (req) => req.reply([]));
+
+    cy.mount(<DeploymentRuleSelector defaultSelector={defaultSelector} onChange={onChange} />);
+}
+
+describe(Cypress.spec.relative, () => {
+    it('Should render "All entities" option when selector is null', () => {
```

Comment:
In cypress/integrations folder:
* Uppercase **Should**: 3 results in 2 files
* lowercase **should**: 559 results in 82 files